"New YAKINDU Statechart" wizard doesn't close after creation of statechart if Project name contains whitespace

This is reproducable by doing the following steps:

1. Create a new General Project with the name "White Space"
2. Create a default Statechart

The Statechart will be created but the "New Yakindu Statechart"-wizard does not close.

The Problem seems to be that the Whitespace in the Project name doesn't get converted properly.

Error message:

```

org.eclipse.emf.ecore.resource.Resource$IOWrappedException: Resource '/White Space/default2.sct' does not exist.
	at org.eclipse.emf.ecore.resource.impl.PlatformResourceURIHandlerImpl$WorkbenchHelper.createPlatformResourceInputStream(PlatformResourceURIHandlerImpl.java:221)
	at org.eclipse.emf.ecore.resource.impl.PlatformResourceURIHandlerImpl.createInputStream(PlatformResourceURIHandlerImpl.java:525)
	at org.eclipse.emf.ecore.resource.impl.ExtensibleURIConverterImpl.createInputStream(ExtensibleURIConverterImpl.java:360)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.load(ResourceImpl.java:1269)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescription.computeExportedObjects(DefaultResourceDescription.java:73)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescription$4.get(DefaultResourceDescription.java:179)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescription$4.get(DefaultResourceDescription.java:173)
	at org.eclipse.xtext.util.OnChangeEvictingCache.get(OnChangeEvictingCache.java:77)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescription.getLookUp(DefaultResourceDescription.java:173)
	at org.eclipse.xtext.resource.impl.AbstractResourceDescription.getExportedObjects(AbstractResourceDescription.java:35)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescriptionManager.isAffected(DefaultResourceDescriptionManager.java:166)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescriptionManager.isAffected(DefaultResourceDescriptionManager.java:146)
	at org.eclipse.xtext.ui.editor.DirtyStateEditorSupport.isReparseRequired(DirtyStateEditorSupport.java:649)
	at org.eclipse.xtext.ui.editor.DirtyStateEditorSupport$UpdateEditorStateJob$1.exec(DirtyStateEditorSupport.java:161)
	at org.eclipse.xtext.ui.editor.DirtyStateEditorSupport$UpdateEditorStateJob$1.exec(DirtyStateEditorSupport.java:1)
	at org.eclipse.xtext.resource.OutdatedStateManager.exec(OutdatedStateManager.java:91)
	at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.internalReadOnly(XtextDocument.java:520)
	at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.readOnly(XtextDocument.java:492)
	at org.eclipse.xtext.ui.editor.model.XtextDocument.readOnly(XtextDocument.java:133)
	at org.eclipse.xtext.ui.editor.DirtyStateEditorSupport$UpdateEditorStateJob.run(DirtyStateEditorSupport.java:146)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
Caused by: org.eclipse.core.internal.resources.ResourceException: Resource '/White Space/default2.sct' does not exist.
	at org.eclipse.core.internal.resources.Resource.checkExists(Resource.java:335)
	at org.eclipse.core.internal.resources.Resource.checkAccessible(Resource.java:209)
	at org.eclipse.core.internal.resources.File.getContents(File.java:275)
	at org.eclipse.core.internal.resources.File.getContents(File.java:268)
	at org.eclipse.emf.ecore.resource.impl.PlatformResourceURIHandlerImpl$WorkbenchHelper.createPlatformResourceInputStream(PlatformResourceURIHandlerImpl.java:207)
	... 20 more
```

--- a/plugins/org.yakindu.sct.generator.core/src/org/yakindu/sct/generator/core/library/AbstractDefaultFeatureValueProvider.java
+++ b/plugins/org.yakindu.sct.generator.core/src/org/yakindu/sct/generator/core/library/AbstractDefaultFeatureValueProvider.java
@@ -70,7 +70,7 @@ public abstract class AbstractDefaultFeatureValueProvider implements IDefaultFea
 	protected String getProjectName(EObject contextElement) {
 		URI uri = EcoreUtil.getURI(contextElement);
 		if (uri.isPlatformResource() && uri.segmentCount() > 1) {
-			return uri.segment(1); // 0 is resource
+			return URI.decode(uri.segment(1)); // 0 is resource
 		}
 		return "ProjectName";
 	}

--- a/plugins/org.yakindu.sct.ui/src/org/yakindu/sct/ui/wizards/ModelCreationWizardPage.java
+++ b/plugins/org.yakindu.sct.ui/src/org/yakindu/sct/ui/wizards/ModelCreationWizardPage.java
@@ -38,7 +38,7 @@ public class ModelCreationWizardPage extends WizardNewFileCreationPage {
 	}
 
 	public URI getURI() {
-		return URI.createPlatformResourceURI(getFilePath().toString(), false);
+		return URI.createPlatformResourceURI(URI.encodeFragment(getFilePath().toString(),true), false);
 	}
 
 	public IPath getFilePath() {

